stack/next.js/render_cache_opt.md

```diff
@@ -0,0 +1,315 @@
+
+-   Busca de dados externos
+
+-   Acesso direto ao banco de dados
+
+-   Utilização de dados sensíveis (Chaves de API, Tokens, ...)
+
+===
+
+=== Client components
+
+-   Uso de `event listeners` (`onClick`, `onChange`)
+
+-   Uso de `state` e `effect` (`useState`, `useEffect`)
+
+-   Uso de APIs do navegador
+
+===
+
+## Caching
+
+!!!
+Tópico não necessário para a produção de aplicações next.
+
+O next apresenta diferentes tipos de `cache` que poderão ser vistos na [documentação](https://nextjs.org/docs/app/building-your-application/caching).
+
+Se o cache for mal feito, a página pode conter erros como dados desatualizados.
+!!!
+
+Por padrão, o next coloca pedaços estáticos de páginas e resultados de `fetch` no `cache`. Contudo, é possível manipular se algo irá para `cache` ou não, quanto tempo permancerá em `cache` e até atualizar o `cache` após certo evento. Para isso, poderão ser utilizados diferentes métodos como `cache()`, `revalidatePath()` e configurações de rota, por exemplo.
+
+### Server Cache
+
+#### Fetch
+
+Resultados recebidos pelo `fetch`, quando utilizado o `server side`, são memorizados até que o componente seja renderizado, por padrão, mas podem ser ajustados passando mais parâmetros para a função.
+
+É possível não gerar cache utilizando:
+
+```tsx page.tsx
+fetch("url", { cache: "no-store" });
+```
+
+É possível "setar" um tempo, em segundos, para revalidar a informação utilizando:
+
+```tsx page.tsx
+fetch("url", { next: { revalidate: 3600 } }); // Revalida a cada 1 hora
+```
+
+### React cache
+
+Se alguma informação faz sentido ser memorizada, mas não é automaticamente, ela pode ir pra cache utilizando a função `cache` do react.
+
+Abaixo tem um exemplo com uma busca no banco de dados:
+
+```ts getItens.ts
+import { cache }
+import prisma from "../prisma/index"
+
+export const getItens = cache(async () => {
+	const itens = await prisma.item.findMany()
+	return itens
+})
+```
+
+### revalidatePath
+
+A função `revalidatePath` é bem interessante quando alguma rota precisa sofrer alguma atualização devido a um evento.
+
+Exemplo com server action:
+
+```tsx page.tsx
+export default async function Page() {
+	const criarItem = async (formData: FormData) => {
+		// Lógica por trás
+
+		revalidatePath("path");
+	};
+
+	const itens = prisma.item.findMany();
+
+	// return da página
+}
+```
+
+### Configurações de rota
+
+Uma rota estática pode ser forçada a ser dinâmica ou pode receber um tempo entre revalidações recebendo "atributos". Para forçar a página a ser dinâmica, é preciso colocar a linha `export const dynamic = "force-dynamic";` e para revalidar, é necessário adicionar `export const revalidate = 3600; // tempo em segundos`.
+
+### Client cache
+
+O next cria um `cache` de rotas no navegador do usuário. Se esse `cache` estiver atrapalhando alguma funcionalidade, poderá ser utilizada a função [`router.refresh()` do hook `useRouter`](https://nextjs.org/docs/app/api-reference/functions/use-router#userouter).
```

Comment:
ainda é melhor usar uma server action que tenha só revalidatePath("path", "page"), como
```
async function refresh() {
  "use server";
  revalidatePath("/path", "page");
}
```
pq ela faz uma revalidação inteligente da rota, sem dar um reload da página, e não precisa forçar um componente a ser client só pra usar o `router.refresh()`

Reply:
não lembro o q eu tava pensando na epoca mas pra mim nao faz mais sentido kk 
esse cache existiria provavelmente por erro na manipulação de cache e a solução certa talvez não seja criar um meio para o usuario sinalizar que precisa de uma atualização ao servidor ou a propria pagina pedir um refresh.